tests/common/include/tests/common/test_geometry_navigation.inl

```diff
@@ -0,0 +1,205 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include <cmath>
+#include <iostream>
+#include <map>
+#include <sstream>
+#include <string>
+#include <utility>
+
+#include "core/detector.hpp"
+#include "core/track.hpp"
+#include "io/csv_io.hpp"
+#include "tools/line_stepper.hpp"
+#include "tools/navigator.hpp"
+#include "utils/ray_gun.hpp"
+
+using namespace detray;
+
+/** Read the detector from file */
+auto read_detector() {
+    auto env_d_d = std::getenv("DETRAY_TEST_DATA_DIR");
+    if (env_d_d == nullptr) {
+        throw std::ios_base::failure(
+            "Test data directory not found. Please set DETRAY_TEST_DATA_DIR.");
+    }
+    auto data_directory = std::string(env_d_d);
+
+    /*std::string name = "odd";
```

Comment:
again ... now I also see your comment, this appears practically everywhere, I hope we can streamline this later.